src/VisualStudio/Core/Def/ProjectSystem/FileChangeWatcher.cs

```diff
@@ -344,6 +375,15 @@ public async ValueTask ApplyAsync(IVsAsyncFileChangeEx service, CancellationToke
                         _token.Cookie = await service.AdviseFileChangeAsync(_directory, _fileChangeFlags, _sink, cancellationToken).ConfigureAwait(false);
                         return;
 
+                    case Kind.WatchFiles:
+                        var cookies = await service.AdviseFileChangesAsync(_files, _fileChangeFlags, _sink, cancellationToken).ConfigureAwait(false);
+
+                        Contract.ThrowIfTrue(cookies.Length != _tokens.Length);
```

Comment:
If this throws, it would mean the cookies that were returned will then be leaked. This really shouldn't happen though, but do we want to have some sort of cleanup in that case? Or at least should we really crash the process since something went very wrong?